### ISSUE TITLE
Should not require a boundary probability map if superpixels are provided in multicut

### ISSUE THREAD

**Issue by k-dominik** [ilastik/ilastik#2115]
## Describe the bug
Right now one has to supply a boundary probability map even if one does not plan to use it for superpixel generation (superpixels generated outside ilastik) and calculating features on it. This is not very intuitive

## Expected behavior
Should be able to proceed in the workflow without providing probability maps

## To Reproduce
Within the ... workflow
1. Add some data in the data selection applet
2. Add some corresponding superpixels

## Desktop (please complete the following information):
 - ilastik version: 1.3.2
 
ilastik ilastik/ilastik@6b143f732f17bf013a84ccfca2577ce6c486f146
lazyflow ilastik/lazyflow@ae5a07647237f50590977f1b3d20cc4af6feef4e
volumina ilastik/volumina@7df7e08bee7d3ee1fcb9f67c8871714adeda2a2b

**Comment by constantinpape** [comment]
This has some implications for the changes @boykovdn is working on:
if we want to make weigths from simple accumulation over the pmap the new default, multicut workflow will need to have a pmap.

Of course, we could introduce some additional logic and only set accumulattion as the default if a pmap if given, but we should at least keep this in the back of our minds.

**Comment by k-dominik** [comment]
That's a good point. it's _boundary based_ segmentation after all. We should definitely decide on how to handle that. I see a huge benefit of simplification, especially when it comes to feature selection in multicut. On the other hand there is also a benefit on applying the multicut on image features...

**Comment by constantinpape** [comment]
> On the other hand there is also a benefit on applying the multicut on image features...

Yeah, I think it's fine to have some logic for optional boundary maps :
```
if have_boundary_maps:
  default_mode = weights_from_accumulation
else:
  default_mode = weight_from_random_forest
  weights_from_accumulation = False  # weights from accumulation not implemented without boundary maps
```

If we decide to do that, we should let @boykovdn know.

> especially when it comes to feature selection in multicut

Feature selection should be simplified to only `BoundaryFeatures` and `RegionFeatures` in any case.
